### ISSUE TITLE
Recheck if upper bound on PyYAML is needed

### ISSUE THREAD

**Issue by matthewfeickert** [reanahub/reana-commons#399]
For reasons that I haven't had time to diagnose yet, installs that limit the upper bound on `PyYAML`, like

https://github.com/reanahub/reana-commons/blob/2a61fffaf9bb2c0f86e3452d0b5c5d4f969c254d/setup.py#L68

are causing install failures

```console
$ docker run --rm -ti python:3.11 /bin/bash
root@f6f2606748d9:/# python -m venv venv && . venv/bin/activate
(venv) root@f6f2606748d9:/# python -m pip --quiet install --upgrade pip setuptools wheel
(venv) root@f6f2606748d9:/# python -m pip list
Package    Version
---------- -------
pip        23.2
setuptools 68.0.0
wheel      0.40.0
(venv) root@f6f2606748d9:/# python -m pip install --upgrade 'PyYAML>=5.1,<6.0'
Collecting PyYAML<6.0,>=5.1
  Downloading PyYAML-5.4.1.tar.gz (175 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 175.1/175.1 kB 1.5 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [68 lines of output]
      /tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/config/setupcfg.py:293: _DeprecatedConfig: Deprecated config in `setup.cfg`
      !!
      
              ********************************************************************************
              The license_file parameter is deprecated, use license_files instead.
      
              By 2023-Oct-30, you need to update your project and remove deprecated calls
              or your builds will no longer be supported.
      
              See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
              ********************************************************************************
      
      !!
        parsed = self.parsers.get(option_name, lambda x: x)(value)
      running egg_info
      writing lib3/PyYAML.egg-info/PKG-INFO
      writing dependency_links to lib3/PyYAML.egg-info/dependency_links.txt
      writing top-level names to lib3/PyYAML.egg-info/top_level.txt
      Traceback (most recent call last):
        File "/venv/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/venv/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/venv/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 338, in run_setup
          exec(code, locals())
        File "<string>", line 271, in <module>
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/__init__.py", line 107, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 1234, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 314, in run
          self.find_sources()
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 322, in find_sources
          mm.run()
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 551, in run
          self.add_defaults()
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 589, in add_defaults
          sdist.add_defaults(self)
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/command/sdist.py", line 104, in add_defaults
          super().add_defaults()
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/sdist.py", line 251, in add_defaults
          self._add_defaults_ext()
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/sdist.py", line 336, in _add_defaults_ext
          self.filelist.extend(build_ext.get_source_files())
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<string>", line 201, in get_source_files
        File "/tmp/pip-build-env-6ud0jyad/overlay/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 107, in __getattr__
          raise AttributeError(attr)
      AttributeError: cython_sources
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
(venv) root@f6f2606748d9:/# 
```

While this is a problem with `PyYAML` and (I think?) ~~`setuptools`~~ `Cython` this is also only being noticed as there is an upper bound on `PyYAML` that was introduced in https://github.com/reanahub/reana-commons/pull/308/commits/ca4fef4a90995b7b559e09dd376e5acd14cbead1 in PR #308.

Is this upper bound still needed? If I remove this upper bound on my fork

```diff
diff --git a/setup.py b/setup.py
index eb8ba53..e29832f 100755
--- a/setup.py
+++ b/setup.py
@@ -65,7 +65,7 @@ install_requires = [
     "jsonschema[format]>=3.0.1,<4.0.0",
     "kombu>=4.6",
     "mock>=3.0,<4",
-    "PyYAML>=5.1,<6.0",
+    "PyYAML>=5.1",
     "Werkzeug>=0.14.1",
     "wcmatch>=8.3,<8.5",
 ]
```

the CI runs without error.

cc @mvidalgarcia (the author of PR #308).

**Comment by matthewfeickert** [comment]
Context: This is https://github.com/yaml/pyyaml/issues/724 and https://github.com/yaml/pyyaml/issues/601 and the reason is as @henryiii pointed out in https://github.com/yaml/pyyaml/issues/724#issuecomment-1638674565 that `Cython` `3.0` is out as of Friday 2023-07-14.

**Comment by mdonadoni** [comment]
Hi @matthewfeickert , thank you for reporting this issue and for opening the related PR!

I hope you don't mind, but I have opened another PR https://github.com/reanahub/reana-commons/pull/401 which bumps the version bounds of PyYAML and already prepares the release commit so that I can directly proceed to publish the new version of reana-commons on PyPI. I have maintaned the lower bound of PyYAML to v5.1, as these older versions still work with Python 3.8/3.9.

Unfortunately PyYAML v6 is not 100% compatible with the rest of REANA. Given that the only stuff that is breaking are some development scripts, and that as far as I can tell we have no option other than upgrading to PyYAML v6.0.1, I think we can proceed with this new release in order to fix at least the installation of reana-client.

Please let me know if this sounds good to you!

**Comment by mdonadoni** [comment]
I have just released 0.9.2.1, so now `reana-commons` and `reana-client` can be installed without issues!
